cpp/src/Ice/Communicator.cpp

```diff
@@ -7,3 +7,26 @@
 Ice::Communicator::~Communicator()
 {
 }
+
+void
+Ice::Communicator::flushBatchRequests(CompressBatch compress)
+{
+    flushBatchRequestsAsync(compress).get();
+}
+
+std::future<void>
+Ice::Communicator::flushBatchRequestsAsync(CompressBatch compress)
+{
+    auto promise = ::std::make_shared<std::promise<void>>();
+    flushBatchRequestsAsync(
+        compress,
+        [promise](::std::exception_ptr ex)
+        {
+            promise->set_exception(::std::move(ex));
```

Comment:
Minor: I thought there was no point on calling std::move with exception_ptr, as per previous reviews, we should also avoid `::std` just `std`

Reply:
That's correct, we should not move exception_ptr. I only "moved" this code around...

Reply:
Fixed